ext/date/php_date.c

```diff
@@ -3705,7 +3705,9 @@ PHP_METHOD(DateTimeZone, __set_state)
 	php_date_instantiate(date_ce_timezone, return_value);
 	tzobj = Z_PHPTIMEZONE_P(return_value);
 	if(php_date_timezone_initialize_from_hash(&return_value, &tzobj, myht) != SUCCESS) {
-		php_error_docref(NULL, E_ERROR, "Timezone initialization failed");
+		zend_throw_error(NULL, "Timezone initialization failed");
+		zval_dtor(return_value);
+		RETURN_FALSE;
```

Comment:
It probably should be simply `return` instead. I used `RETURN_FALSE` because it was used if `zend_parse_parameters()` fails. I can change both occurrences.